src/nxt_http_static.c

```diff
@@ -368,7 +368,7 @@ nxt_http_static_send_ready(nxt_task_t *task, void *obj, void *data)
             }
         }
 
-        fname = ctx->share.start;
+        fname = shr->start;
     }
 
     nxt_memzero(&file, sizeof(nxt_file_t));
```

Comment:
I thought mixing changes `shr`->`path` and `ctx->share`->`path` might be confusing, and might let a future reviewer wondering why `shr` was used in some places and `ctx->share` in another, and if I made a mistake.

But yeah, maybe that's not too much information together and they ccan be squashed together.  As you prefer.  Both ways are good to me.

---

Since that's the only `ctx->share` use in the function after setting `shr`, it was probably a leftover of Zhidao's refactor.

Reply:
Given that in the first patch you do (as the only change)

```diff
-        fname = ctx->share.start;
+        fname = shr->start;
```

Then in the second patch

```diff
-        fname = shr->start;
     }
 
+    fname = path->start
```

I.e fname went from ```= shr->start``` to ```= path->start```

The first seems superfluous.

Reply:
Yeah, mostly.  I'll squash them.